java/Cherepnin_Oleh/src/com/zoolatech/lecture3/tasks/_2/NameValidator.java

```diff
@@ -0,0 +1,30 @@
+package com.zoolatech.lecture3.tasks._2;
+
+import java.lang.reflect.Field;
+
+public class NameValidator implements SmallerValidator {
```

Comment:
You needed to define a class for each validation rule:

```
List of validation rules: string is not empty, valid email, valid phone number, number is in the range between X and Y, number is bigger than a value X, number is less than a value X. 
```

Reply:
> number is in the range between X and Y, number is bigger than a value X, number is less than a value X.
Should I split DateValidator class into several other?

Reply:
Yes. You don't need to implement a validator for a specific field/fields. You need to implement validators for specific validation rules.